notebooks/concepts/Specviz from splot concept notebook.ipynb

```diff
@@ -0,0 +1,3636 @@
+    "\n",
+    "1. For notebooks, providing an easy way to extract SpectralRegion objects from selections made in the specviz UI. This would most straightforwardly be accomplished by making a subset selection, and then providing a way to convert that into a SpectralRegion.  E.g., if the user uses the horizontal-selection to create a subset,  they can do:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "reg = specviz.get_selected_region()  #<- uses the \"most recent\" selection\n",
+    "reg2 = specviz.get_selected_region('Subset 2')  #<- uses glue's name for the subset\n",
+    "\n",
+    "all_regs = specviz.get_selected_region(all=True)  #<- this would be *all* the subsets combined together into a single SpectralRegion object, which is [ALREADY IMPLEMENTED] by SpectralRegion (but not specviz)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "    Note that it is critical for some use cases that the above should yield spectral regions that are correct *even to the sub-pixel*.  That is, if the user does a region selection that ends between two pixels, that should be reflected in the SpectralRegion - this means using subset *states* in glue instead of the subset itself.\n",
+    "\n",
+    "    These selections can now be used with standard specutils analysis functions in a notebook setting (these are [ALREADY IMPLEMENTED]):"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "specutils.analysis.line_flux(specviz.current_spectrum, specviz.get_selected_region())\n",
+    "\n",
+    "specutils.analysis.equivalent_width(specviz.current_spectrum, regions=specviz.get_selected_region())\n",
+    "\n",
+    "specutils.analysis.gaussian_fwhm(specviz.current_spectrum, specviz.get_selected_region())\n",
+    "\n",
+    "specutils.analysis.snr(specviz.current_spectrum, specviz.get_selected_region())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "    In addition to region selection, there is value in having a way to extract a specific point in the displayed spectrum into the notebook. (Some examples are shown below) - while there are many potential solutions to this, the simplest is a stateful \"get location\" method.  E.g.,"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "spectral_coordinate, flux_coordinate = specviz.get_spectrum_location()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "    This method would block, cause the cursor in the viz tool to change to some indicator cursor, and then un-block as soon as the user clicks somewhere in the viewer window, reporting out the spot the click."
```

Comment:
This smells like a splot-ism that isn't really cohesive with modern application behavior. My first concern is that viewer tools are not global; they only operate within their viewer. Changing the user interaction globally will require re-evaluating how viewer tools operate in glue-jupyter (and how they interface with jupyter notebook environments). 

Why not just use a pixel select subset mode and have the user use `get_selected_region('region 3')`? Alternatively, have a tool that does a position measurement in the application and let the user call `specviz.get_marked_location()` to return the stored value.

Reply:
Good points. I put them all in as *options*, but I think I like your "marking" approach best, and will update the notebook accordingly.